w3w-autosuggest/admin/js/w3w-autosuggest-admin.js

```diff
@@ -0,0 +1,271 @@
+    clipToBBCheckbox.addEventListener('change', function(e) {
+      handleToggleTarget(clipToBBSWLatInput, e.target.checked)
+      handleToggleTarget(clipToBBSWLngInput, e.target.checked)
+      handleToggleTarget(clipToBBNELatInput, e.target.checked)
+      handleToggleTarget(clipToBBNELngInput, e.target.checked)
+    })
+  }
+
+  /**
+   * CLIP TO POLYGON ADVANCED HANDLER
+   */
+  const clipToPolygonCheckbox = document.querySelector('input#enable_clip_to_polygon')
+  const clipToPolygonTextarea = document.querySelector('textarea#clip_to_polygon')
+  if (clipToPolygonCheckbox && clipToPolygonTextarea) {
+    handleToggleTarget(clipToPolygonTextarea, clipToPolygonCheckbox.checked)
+    clipToPolygonCheckbox.addEventListener('change', function(e) {
+      handleToggleTarget(clipToPolygonTextarea, e.target.checked)
+    })
+  }
+
+  /**
+   * SETTINGS FORM HANDLER
+   */
+  const settingsForm = document.querySelector('form#settings')
+  if (settingsForm) {
+    settingsForm.addEventListener('submit', function(e) {
+      e.target.classList.remove('was-validated')
+
+      if (convertExistingRadio.checked && !validateTarget(inputSelectorInput, function(value) {
+        return value.length > 0 && (value.indexOf('#') > -1 || value.indexOf('.') > -1)
+      })) return e.preventDefault()
+
+      if (showLabelCheckbox.checked && !validateTarget(labelInput, function(value) {
+        return value.length > 0
+      })) return e.preventDefault()
+
+      e.target.classList.add('was-validated')
+    })
+  }
+
+  /**
+   * ADVANCED SETTINGS FORM HANDLER
+   */
+  const advancedForm = document.querySelector('form#advanced')
+  if (advancedForm) {
+    advancedForm.addEventListener('submit', function(e) {
+      e.target.classList.remove('was-validated')
+
+      if (placeholderCheckbox.checked && !validateTarget(placeholderInput, function(value) {
+        return value.trim().length > 0
+      })) {
+        placeholderInput.scrollTo()
+        return e.preventDefault()
+      }
+
+      if (clipToCountryCheckbox.checked && !validateTarget(clipToCountryInput, function(value) {
+        return value.split(',').filter(val => val.trim().length !== 2 || !(/^[A-Z]{2}$/.test(val.trim()))).length === 0
+      })) {
+        clipToCountryInput.scrollTo()
+        return e.preventDefault()
+      }
+
+      if (clipToCircleCheckbox.checked) {
+        if (!validateTarget(clipToCircleLatInput, function(value) {
+          return COORD_REGEX.test(value.trim())
+        })) {
+          clipToCircleLatInput.scrollTo()
+          return e.preventDefault()
+        }
+        if (!validateTarget(clipToCircleLngInput, function(value) {
+          return COORD_REGEX.test(value.trim())
+        })) {
+          clipToCircleLngInput.scrollTo()
+          return e.preventDefault()
+        }
+        if (!validateTarget(clipToCircleRadiusInput, function(value) {
+          const parsedValue = /\d+/.test(value.trim())
+          return parsedValue !== NaN
```

Comment:
Would be safer to use [isNaN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/isNaN) instead - may result in bugs with this (unless that logic is intended)

Reply:
This is actually incorrect, I should just return the boolean value of the `regex.test` function.